### PR TITLE
Adding a min width and a min height for the main window

### LINKED ISSUE #9606: Set a minHeight and a minWidth for the main window
**Is your suggestion for improvement related to a problem? Please describe.**
The JabRef main window has no minimum width or height. So you can decrease the 
size until it can be no longer seen. Which I think is not a good thing. 
(See 'Window before change' in the **Additional context** below &darr;)

**Describe the solution you'd like**
I simply suggest to add a `setMinHeight` and a `setMinWidth` in the `openWindow()` method under
**JabRefGui** class.
code-after-edit

I have chosen the values of `minWidth` and `minHeight` as follows:
- `minWidth`:
  - window-with
  - I've chosen the `width` in a way that the up menu bar keeps displaying all its 
  content (**File**, **Edit**, **...**) after resize (after decreasing the size).
  - It's about **580px** (Using Ubuntu 20.04).

- `minHeight`:
  - min-height
   - I've chosen the `height` in a way that the element **OpenOffice/LibreOffice** is shown entirely after resize (after decreasing the size).
   I chosen **OpenOffice/LibreOffice** as a reference, because I found that it has the highest height among other elements under 
  the **view** option menu (compared to **Web search** and **Groups**).
- It's about **330px** (Using Ubuntu 20.04).

**Additional context**
- Window before change
jabref-window-fix

- Minimum window size after change
jabref-window-after-change

**Other**
Looking forward to hear your feedback, and to make a PR on this.

### PR DESCRIPTION
**Pull request by waterflow80**
### PR Description
Fixes #9606 

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
Hello @waterflow80 , thanks for your interest in JabRef and open source programming. Can you provide some more information on your changes please? What is your motivation for your change, is there an issue related to your changes, why would be a minimum height oder width necessary, why exactly 330 px? Thanks!

**Comment by calixtus** [comment]
Also please note that the checklist is mandatory.

**Comment by waterflow80** [comment]
@calixtus Thank you for your feedback.

I think having the app's window resized until it's no longer seen is not a good thing, as it's shown in the 
GIF image in the  issue #9608 description. And almost every app has a minimum window size.

I also specified the reasons why I've chosen the values of the `minHeight` and the `minWidth` in the issue #9608, although 
these values can be discussed or reviewed.

File: src/main/java/org/jabref/gui/JabRefGUI.java [hunk 1]
```diff
@@ -68,6 +68,11 @@ private void openWindow(Stage mainStage) {
         mainFrame.init();
 
         GuiPreferences guiPreferences = preferencesService.getGuiPreferences();
+
+        // Set the min-width and min-height for the main window
```
Comment (HoussemNasri) [inline src/main/java/org/jabref/gui/JabRefGUI.java:72]: I don't think the comment is necessary

Comment (waterflow80) [reply src/main/java/org/jabref/gui/JabRefGUI.java:72]: @HoussemNasri Thank you for your feedback, I'll remove it if necessary.

**Review by koppor — APPROVED** [review]
Discussed in DevCall. It is OK. More comments are in the issue.

**Pull request merged**